exporter/alertmanagerexporter/alertmanager_exporter.go

```diff
@@ -23,20 +33,171 @@ type alertmanagerExporter struct {
 	severityAttribute string
 }
 
-func (s *alertmanagerExporter) pushTraces(_ context.Context, _ ptrace.Traces) error {
+type alertmanagerEvent struct {
+	spanEvent ptrace.SpanEvent
+	traceID   string
+	spanID    string
+	severity  string
+}
+
+func (s *alertmanagerExporter) convertEventSliceToArray(eventSlice ptrace.SpanEventSlice, traceID pcommon.TraceID, spanID pcommon.SpanID) []*alertmanagerEvent {
+	if eventSlice.Len() > 0 {
+		events := make([]*alertmanagerEvent, eventSlice.Len())
+
+		for i := 0; i < eventSlice.Len(); i++ {
+			var severity string
+			severityAttrValue, ok := eventSlice.At(i).Attributes().Get(s.severityAttribute)
+			if ok {
+				severity = severityAttrValue.AsString()
+			} else {
+				severity = s.defaultSeverity
+			}
+			event := alertmanagerEvent{
+				spanEvent: eventSlice.At(i),
+				traceID:   traceID.String(),
+				spanID:    spanID.String(),
+				severity:  severity,
+			}
+
+			events[i] = &event
+		}
+		return events
+	}
+	return nil
+}
+
+func (s *alertmanagerExporter) extractEvents(td ptrace.Traces) []*alertmanagerEvent {
+
+	// Stitch parent trace ID and span ID
+	rss := td.ResourceSpans()
+	var events []*alertmanagerEvent
+	if rss.Len() == 0 {
+		return nil
+	}
+
+	for i := 0; i < rss.Len(); i++ {
+		resource := rss.At(i).Resource()
+		ilss := rss.At(i).ScopeSpans()
+
+		if resource.Attributes().Len() == 0 && ilss.Len() == 0 {
+			return nil
+		}
+
+		for j := 0; j < ilss.Len(); j++ {
+			spans := ilss.At(j).Spans()
+			for k := 0; k < spans.Len(); k++ {
+				traceID := spans.At(k).TraceID()
+				spanID := spans.At(k).SpanID()
+				events = append(events, s.convertEventSliceToArray(spans.At(k).Events(), traceID, spanID)...)
+			}
+		}
+	}
+	return events
+}
+
+func createAnnotations(event *alertmanagerEvent) model.LabelSet {
+	labelMap := make(model.LabelSet, event.spanEvent.Attributes().Len()+1)
```

Comment:
```suggestion
	labelMap := make(model.LabelSet, event.spanEvent.Attributes().Len()+2)
```
Since you add both `TraceID` and `SpanID`